snippets/auth.go

```diff
@@ -379,3 +381,196 @@ func listUsers(ctx context.Context, client *auth.Client) {
+	}
+	// [END import_with_pbkdf]
+}
+
+func importWithStandardScrypt(ctx context.Context, client *auth.Client) {
+	// [START import_with_standard_scrypt]
+	users := []*auth.UserToImport{
+		(&auth.UserToImport{}).
+			UID("some-uid").
+			Email("user@example.com").
+			PasswordHash([]byte("password-hash")).
+			PasswordSalt([]byte("salt")),
+	}
+	h := hash.StandardScrypt{
+		MemoryCost:       1024,
+		Parallelization:  16,
+		BlockSize:        8,
+		DerivedKeyLength: 64,
+	}
+	result, err := client.ImportUsers(ctx, users, auth.WithHash(h))
+	if err != nil {
+		log.Fatalln("Error importing users", err)
+	}
+	for _, e := range result.Errors {
+		log.Println("Failed to import user", e.Reason)
+	}
+	// [END import_with_standard_scrypt]
+}
+
+func importWithBcrypt(ctx context.Context, client *auth.Client) {
+	// [START import_with_bcrypt]
+	users := []*auth.UserToImport{
+		(&auth.UserToImport{}).
+			UID("some-uid").
+			Email("user@example.com").
+			PasswordHash([]byte("password-hash")).
+			PasswordSalt([]byte("salt")),
+	}
+	h := hash.Bcrypt{}
+	result, err := client.ImportUsers(ctx, users, auth.WithHash(h))
+	if err != nil {
+		log.Fatalln("Error importing users", err)
+	}
+	for _, e := range result.Errors {
+		log.Println("Failed to import user", e.Reason)
+	}
+	// [END import_with_bcrypt]
+}
+
+func importWithScrypt(ctx context.Context, client *auth.Client) {
+	// [START import_with_scrypt]
+	users := []*auth.UserToImport{
+		(&auth.UserToImport{}).
+			UID("some-uid").
+			Email("user@example.com").
+			PasswordHash([]byte("password-hash")).
+			PasswordSalt([]byte("salt")),
+	}
+	b64decode := func(s string) []byte {
+		b, err := base64.StdEncoding.DecodeString(s)
+		if err != nil {
+			log.Fatalln("Failed to decode string", err)
+		}
+		return b
+	}
+
+	// All the parameters below can be obtained from the Firebase Console's "Users"
+	// section. Base64 encoded parameters must be decoded into raw bytes.
+	h := hash.Scrypt{
+		Key:           b64decode("base64-secret"),
+		SaltSeparator: b64decode("base64-salt-separator"),
+		Rounds:        8,
+		MemoryCost:    14,
+	}
+	result, err := client.ImportUsers(ctx, users, auth.WithHash(h))
+	if err != nil {
+		log.Fatalln("Error importing users", err)
+	}
+	for _, e := range result.Errors {
+		log.Println("Failed to import user", e.Reason)
+	}
+	// [END import_with_scrypt]
+}
+
+func importWithoutPassword(ctx context.Context, client *auth.Client) {
+	// [START import_without_password]
+	users := []*auth.UserToImport{
+		(&auth.UserToImport{}).
+			UID("some-uid").
+			DisplayName("John Doe").
+			Email("johndoe@gmail.com").
+			PhotoURL("http://www.example.com/12345678/photo.png").
+			EmailVerified(true).
+			PhoneNumber("+11234567890").
+			CustomClaims(map[string]interface{}{"admin": true}). // set this user as admin
+			ProviderData([]*auth.UserProvider{                   // user with Google provider
+				{
+					UID:         "google-uid",
+					Email:       "johndoe@gmail.com",
```

Comment:
We probably want this to be @example.com.

Reply:
The Node.js sample already published by the Auth team uses the `gmail` domain. This is just being consistent with that.

https://firebase.google.com/docs/auth/admin/import-users#import_users_without_passwords